java/com/tigervnc/vncviewer/OptionsDialog.java

```diff
@@ -1173,31 +1179,54 @@ private JPanel createMiscPanel() {
     MiscPanel.setBorder(BorderFactory.createEmptyBorder(5, 5, 0, 5));
     sharedCheckbox =
       new JCheckBox("Shared (don't disconnect other viewers)");
-    dotWhenNoCursorCheckbox = new JCheckBox("Show dot when no cursor");
+    alwaysCursorCheckbox = new JCheckBox("Show local cursor when not provided by server");
+    alwaysCursorCheckbox.addItemListener(new ItemListener() {
+      public void itemStateChanged(ItemEvent e) {
+        handleAlwaysCursor();
+      }
+    });
+    JLabel cursorTypeLabel = new JLabel("Cursor type:");
+    String[] cursorTypes = {"Dot", "System"};
+    cursorTypeChoice = new MyJComboBox(cursorTypes);
+    cursorTypeChoice.setEditable(true);
```

Comment:
Why is this editable?  Doing so allows for an arbitrary string to entered but then subsequently only checks for "Dot" or "System".

Reply:
Because I got setEditable confused with setEnabled and didn't notice the difference in behavior when testing. I've corrected it now.